commands/add.go

```diff
@@ -2,64 +2,63 @@ package commands
 
 import (
 	"fmt"
-	"os"
 
 	"github.com/bndw/pick/errors"
 	"github.com/bndw/pick/utils"
 	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
 )
 
 func init() {
 	rootCmd.AddCommand(&cobra.Command{
 		Use:   "add [name] [username] [password]",
 		Short: "Add a credential",
-		Long: `The add command is used to add a new credential.
-            `,
+		Long:  "The add command is used to add a new credential.",
 		Run: func(cmd *cobra.Command, args []string) {
-			os.Exit(Add(args...))
+			runCommand(Add, cmd, args)
 		},
 	})
 }
 
-func Add(args ...string) int {
-	safe, err := loadSafe()
+func Add(args []string, flags *pflag.FlagSet) error {
+	safe, err := newSafeLoader().LoadSafe()
 	if err != nil {
-		return handleError(err)
+		return err
 	}
 
-	name, username, password, errCode := parseAddArgs(args)
-	if errCode > 0 {
-		return errCode
+	name, username, password, err := parseAddArgs(args)
+	if err != nil {
+		return err
 	}
 
 	account, err := safe.Add(name, username, password)
 	if _, conflict := err.(*errors.AccountExists); conflict && overwrite(name) {
 		var editErr error
 		if account, editErr = safe.Edit(name, username, password); editErr != nil {
-			return handleError(editErr)
+			return editErr
 		}
 	} else if err != nil {
-		return handleError(err)
+		return err
 	}
 
 	fmt.Println("Credential added")
 	if utils.Confirm("Copy password to clipboard", true) {
 		if err := utils.CopyToClipboard(account.Password); err != nil {
-			return handleError(err)
+			return err
 		}
 	}
-	return 0
+	return nil
 }
 
 func overwrite(name string) bool {
 	prompt := fmt.Sprintf("%s already exists, overwrite", name)
 	return utils.Confirm(prompt, false)
 }
 
-func parseAddArgs(args []string) (name, username, password string, errCode int) {
+func parseAddArgs(args []string) (name, username, password string, err error) {
 	if len(args) > 3 {
-		fmt.Println("Usage: add [name] [username] [password]")
-		return "", "", "", 1
+		err = &errors.InvalidCommandUsage{}
```

Comment:
Not specific to this location, but we should consider instantiating some error variables to cleanup the call sites.

```go
package errors

var (
  InvalidUsageErr  = ...
)
```
```go
package foo

...
if len(args) > 3 {
  return errors.ErrInvalidUsage
}
```